.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,27 @@
+name: Go
+
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    branches:
+      - '**'
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '^1.22'
+
+    - name: Install dependencies
+      run: go mod tidy
+
+    - name: Run tests
+      run: go test ./...
```

Comment:
Great addition!

just re-reading this now, but this doesn't look like e2e tests isn't it? This is actually covering #369  . Please rename the pipeline file accordingly, would be even better to keep in a separate PR as doesn't seems we are bringing new tests in this PR that are covered by it?

```

Run go test ./...
?   	github.com/masa-finance/masa-oracle/cmd/masa-node	[no test files]
?   	github.com/masa-finance/masa-oracle/cmd/masa-node-cli	[no test files]
?   	github.com/masa-finance/masa-oracle/cmd/tagger	[no test files]
?   	github.com/masa-finance/masa-oracle/docs	[no test files]
?   	github.com/masa-finance/masa-oracle/pkg/api	[no test files]
?   	github.com/masa-finance/masa-oracle/pkg/chain	[no test files]
?   	github.com/masa-finance/masa-oracle/pkg/config	[no test files]
ok  	github.com/masa-finance/masa-oracle/pkg	0.026s
?   	github.com/masa-finance/masa-oracle/pkg/db	[no test files]
?   	github.com/masa-finance/masa-oracle/pkg/llmbridge	[no test files]
ok  	github.com/masa-finance/masa-oracle/pkg/consensus	0.015s
?   	github.com/masa-finance/masa-oracle/pkg/network	[no test files]
?   	github.com/masa-finance/masa-oracle/pkg/pubsub	[no test files]
?   	github.com/masa-finance/masa-oracle/pkg/scrapers/discord	[no test files]
?   	github.com/masa-finance/masa-oracle/pkg/scrapers/reddit	[no test files]
?   	github.com/masa-finance/masa-oracle/pkg/scrapers/twitter	[no test files]
?   	github.com/masa-finance/masa-oracle/pkg/scrapers/web	[no test files]
?   	github.com/masa-finance/masa-oracle/pkg/staking	[no test files]
ok  	github.com/masa-finance/masa-oracle/pkg/masacrypto	0.016s
?   	github.com/masa-finance/masa-oracle/pkg/workers	[no test files]
?   	github.com/masa-finance/masa-oracle/pkg/workers/messages	[no test files]
?   	github.com/masa-finance/masa-oracle/tests	[no test files]
?   	github.com/masa-finance/masa-oracle/tools	[no test files]
ok  	github.com/masa-finance/masa-oracle/pkg/tests	1.306s
```